docs/pages/blog/masonry-post.md

```diff
@@ -0,0 +1,141 @@
+
+The property `column-gap` should change in response to the `spacing` prop. For example, if `spacing` is `2`, `<Masonry />` will have the following property: `column-gap: theme.spacing(2)`. It is worth noting here that `spacing` is a factor of the theme's spacing. You may ask why we only change the `column-gap`. What about the `row-gap`? As we already discussed, `row-gap: 1` should always be held in order for this implementation to work. Otherwise, `<MasonryItem />` cannot span in a multitude of `1px`.
+
+This is pretty much the gem of `<Masonry />`. It is much simpler relative to `<MasonryItem />`, which we are now moving on to look in detail.
+
+### `<MasonryItem />`
+
+This component is a CSS grid item that wraps around one and only child. Through the use of `React.Children.only()`, we enforce that only one child is passed to the component. MUI users can always create more `<MasonryItem />` to add more children.
+
+`<MasonryItem />` receives two special props: `columnSpan` and `defaultHeight`. `columnSpan` is the number of columns taken up by the component, while `defaultHeight`, which is provided for optional server-side rendering, is the initial height of the component. Both props receive a fixed value. Further details can be found in its [API documents](https://next.material-ui.com/api/masonry-item/).
+
+`<MasonryItem />` utilises a Web API called [ResizeObserver API](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver). It detects changes to the dimensions of an element. We use this API to observe the child passed to `<MasonryItem />`. Every time the child's height changes, which is commonly caused by window resizing or zooming, these two properties, `grid-row-end` and `padding-bottom`, also get updated. `grid-row-end` changes so that the height of `<MasonryItem />`, by spanning a correct number of rows, matches its child's height. `padding-bottom` changes so that the vertical gap between `<MasonryItem />`s stays correct.
+
+If user wants to use the column spanning feature and therefore passes `columnSpan` prop to the component, the property `grid-column-end` is assined the given value. For example, if `columnSpan` is `3`, `grid-column-end` is set to `span 3` and the component will span three columns.
+
+If user wants to use server-side rendering and hence passes `defaultHeight` prop to the component, ResizeObserver API is not used so that the component, `<MasonryItem />`, is set to the given height and retains it. However, if `defaultHeight` is larger than the height of its child, there will be unwanted gap between the component and the child. If it is less, some parts of the child will not be displayed. Hence, the prop `defaultHeight` should be used with special attention to enjoy both benefits of a solid masonry and server-side rendering.
```

Comment:
```suggestion
If the developer wants to use server-side rendering and so passes the `defaultHeight` prop to the component, the ResizeObserver API is not used so that the component, `<MasonryItem />`, is set to the given height and retains it. However, if `defaultHeight` is larger than the height of its child, there will be an unwanted gap between the component and the child. If it is less, some parts of the child will not be displayed. Hence, the prop `defaultHeight` should be used with special attention to enjoy the benefits of both a solid masonry and server-side rendering.
```